docs/workflow/continuous_integration.rst

```diff
@@ -0,0 +1,181 @@
+
+Copy the saved known host, as we will need it for installing the key on azure.
+It should look something like (should look something like [1]As3..=ssh-rsa ..) and will be the last line added to::
+
+     ~/.ssh/known_hosts 
+
+
+Setting up azure services 
+=========================
+
+The first step is to visit `azure devops`_ and create an account.
+
+Adding the key to azure's secure files
+--------------------------------------
+
+Upload the secure key to azure's library to download it in the script.
+
+In the library tab, select the **Secure files** tab, upload the file, and authorize it for use in all pipelines.
+As seen here:
+  .. image:: images/secure_file.png
+
+See also: `azure secure files`_   
+
+One must use the DownloadSecureFile@1 `azure task`_ to download the file onto the virtual machine when the pipeline runs.
+Afterwhich, one must use the InstallSSHKey@0 `azure task`_ to add the ssh key.
+
+Setting up the YAML file to deploy
+----------------------------------
+
+YAML is short for "YAML Ain't Markup Language", as it is intented to be a simple way to write script
+that is standard for all programing languages. It is the file that communicates directly with the
+pipeline. 
+
+It follows the following hierarchy:
+
+Pipeline hierarchy
+^^^^^^^^^^^^^^^^^^
+
+.. graphviz::
+
+  digraph {
+    a -> b -> c -> d -> e
+    d -> f
+    a [label="Stages",shape=box,fillcolor="white",style="filled"];
+    b [label="Stage",shape=circle,fillcolor="white",style="filled"];
+    c [label="Jobs", shape="circle", fillcolor="white", style="filled"]
+    d [label="Job", shape="circle", fillcolor="white", style="filled"]
+    e [label="Steps", shape="circle", fillcolor="white", style="filled"]
+    f [label="Task", shape="circle", fillcolor="white", style="filled"]
+  }
+
+A job is referred to as the most basic building block that the pipeline lines, which is composed of steps of script.
+A task is a predefined script with a definitive purpose, such as downloading the secure file.
+
+To download and add the ssh-key, prepare the scripts as::
+
+      - task: DownloadSecureFile@1
+        inputs: 
+          secureFile: 'id_azure_rsa'
+
+Secure files stored in the azure server are encryped and again decrypted by the azure task that uses the file.
+
+Download a secure file to a temporary location in the virtual machine::
+
+      - task: InstallSSHKey@0
+        inputs:
+          hostName: $(gh_host)
+          sshPublicKey: $(public_key)
+          #sshPassphrase: # Optional - leave empty if it was left empty while generating the key.
+          sshKeySecureFile: 'id_azure_rsa'
+
+
+hostName is the line that was copied at the end of `Setting up a secure pathway to github`.
+
+sshPublicKey should be a string value of what is inside your .pub file (i.e: rsa-key Axddd... username@server).
+
+sshKeySecureFile is the downloaded secure file you generated, you can reference directly as shown.
+
+
+To define variables in the script, one can do so as::
```

Comment:
I'm sorry to say, but at 1st look, it's appearing to me, as if this definition of `system.debug` is per-requisite for defining variables. (A subjective opinion)

I would suggest to rephrase it as:
`Variables can be defined in the script, simply as key-value pairs. For example::`

And I must say, you've articulated the entire process really great, and your use of images, make things even clearer!